components/script/script_thread.rs

```diff
@@ -3082,20 +3073,20 @@ impl ScriptThread {
         &self,
         global_to_clone: &GlobalScope,
         top_level_browsing_context_id: TopLevelBrowsingContextId,
-        pipeline_id: PipelineId,
+        browsing_context_id: BrowsingContextId,
+        parent: Option<BrowsingContextId>,
         opener: Option<BrowsingContextId>,
     ) -> Option<DomRoot<WindowProxy>> {
-        let (browsing_context_id, parent_pipeline_id) =
-            self.ask_constellation_for_browsing_context_info(pipeline_id)?;
         if let Some(window_proxy) = self.window_proxies.borrow().get(&browsing_context_id) {
             return Some(DomRoot::from_ref(window_proxy));
         }
 
-        let parent_browsing_context = parent_pipeline_id.and_then(|parent_id| {
+        let parent_browsing_context = parent.and_then(|parent_bc| {
             self.remote_window_proxy(
                 global_to_clone,
                 top_level_browsing_context_id,
-                parent_id,
+                parent_bc,
+                None,
```

Comment:
I don't understand why this is `None`, can't there be a grandparent?

Reply:
I'm not completely sure, but my impression was that although that function is written in a recursive style, it's not actually used recursively beyond one level. Maybe it's worth writing a test in `mozilla` for.

Reply:
Ah no, looking at the docs for it, the intent is definitely for it to be recursive...

Reply:
Yes, it should definitely be recursive!